### PR TITLE
Shiny Apps Deployment

### PR DESCRIPTION
**Pull request by brucehoff**
This PR adds a workflow to deploy to ShinyApps.  Work in progress is deployed to `/data_curator-staging`.  When a `release*` branch is created the app will be deployed to `/data_curator`.  This allows us to develop and test while the production version continues to run.

Besides adding the workflow, this PR:

- modifies `global.R` to unzip and activate the virtual environment.
- modifies `install-pkgs.R` to add some R packages that seem to be needed and to use R Studio's binary package distribution, which dramatically speeds up the build.
- modifies `server.R` to correctly use the OAuth access token obtained during use login.  The existing app' wrongly uses the Synapse access token from the browser cookie which (1) gives the app' unlimited "scope" in accessing the user's Synapse account and (2) will only work when the app' is hosted in the `*.synapse.org` domain.  This change fixes both problems.  The app' can be hosted anywhere!

Before merging the PR, set the following parameters in the data_curator GitHub repo' (under Settings > Secrets > New repository secret):

- REPO_PAT: This is a GitHub personal access token having (at least) 'repository' scope.
- RSCONNECT_USER: sagebio
- RSCONNECT_TOKEN: See the `token` in the `shinyapps.io` record in LastPass.
- RSCONNECT_SECRET:  See the `secret` in the `shinyapps.io` record in LastPass.
- OAUTH_CLIENT_ID: Use 100132 or create your own.
- OAUTH_CLIENT_SECRET: If you use 100132, retrieve the secret from the `ShinyApps Synapse account` record in LastPass.

### CODE REVIEW

**Comment by milen-sage** [comment]
Thanks @brucehoff I added the secrets you noted above to the repo and am going through the PR; using oauth client id 100132 while we are testing.

File: .github/workflows/shinyapps_deploy.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+          chmod 755 "${{ env.CONDA_ENV_NAME }}"/bin/activate
+          source "${{ env.CONDA_ENV_NAME }}"/bin/activate
+        
+      - name: Install Python Dependencies
+        shell: bash
+        run: |
+          source "${{ env.CONDA_ENV_NAME }}"/bin/activate
+          # install 'poetry' and then use it to install 'schematic'
+          pip3 install poetry
+          git clone --single-branch --branch develop https://github.com/Sage-Bionetworks/schematic.git
```
Comment (milen-sage) [inline .github/workflows/shinyapps_deploy.yml:47]: @afwillia a note here: this assumes building from the develop branch of schematic. It's fine for testing. Longer term we want to use a release branch of schematic; or, switching to REST API and avoiding installation of schematic, whichever happens first :)

File: .github/workflows/shinyapps_deploy.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          # No such file or directory: 'tests/data/example.model.jsonld'
+          mkdir -p ../tests/data/
+          cp -R tests/data/ ../tests/
+        
+      - name: Install R packages
+        run: |
+          R -f install-pkgs.R
+          
+      - name: zip virtual env
+        # ShinyApps has a limit of 7000 files, far exceeded by the many Python dependencies
```
Comment (milen-sage) [inline .github/workflows/shinyapps_deploy.yml:64]: @brucehoff thanks for pointing out this workaround. The dependencies in python quickly add up, but point taken. We will 1) in general, ensure only the dependencies we need are part of schematic's setup; and 2) switch code organization using python subpackages to be better able to control schematic's setup. In the context of the DC app specifically, we will transition away from local installation of schematic altogether, and switch to endpoints exposing schematic's functionality via REST API.

**Review by milen-sage — APPROVED** [review]
Thanks @brucehoff! Will quickly review with @rrchai and @afwillia tomorrow as well. We'll make the couple of changes related to credential files and continue testing. The PR looks good to me!

**Comment by brucehoff** [comment]
Thanks for the review, @milen-sage .  I do not have permission to merge the PR so could you either grant me permission or merge the PR yourself?  Thank you.

**Pull request merged**